tests/unit/restapi/test_db_models.py

```diff
@@ -907,3 +911,57 @@ def test_db_invalid_resource_dependency_fails(
         except IntegrityError as err:
             db.session.rollback()
             raise err
+
+
+def test_task_engine_yaml_creation_works(
+    db: SQLAlchemy,
+    registered_plugin_id: int,
+    registered_entry_point_id: int,
+) -> None:
+    entrypoint = viewsdb.get_latest_entry_point(
+        db, resource_id=registered_entry_point_id
+    )
+    plugin = viewsdb.get_latest_plugin(db, resource_id=registered_plugin_id)
+    files = viewsdb.get_latest_plugin_files(db, plugin_resource_id=registered_plugin_id)
+
+    types = {
+        param.parameter_type.name: param.parameter_type.structure
+        for file in files
+        for task in file.tasks
+        for param in task.input_parameters + task.output_parameters
+        if param.parameter_type.name not in BUILTIN_TYPES
+    }
+
+    parameters = {param.name: param.default_value for param in entrypoint.parameters}
+
+    tasks = {
+        task.plugin_task_name: {
+            "plugin": ".".join(
+                [plugin.name, Path(file.filename).stem, task.plugin_task_name]
+            ),
```

Comment:
I think this section will require a bit more logic, since it doesn't allow for subfolders. Say we have `utils/raw/sample.py` as the filename (I'm realizing now that may not be permitted in the Marshmallow schema, so I'll need to adjust it if that's not allowed). We could do something like the following to construct the plugin section:

```python
".".join(
    [
        plugin.name,
        *[Path(x).stem] for x in Path("utils/raw/sample.py").parts],
        task.plugin_task_name,
    ]
)
```

Although I don't think we should rely on `Pathlib` here, since that means this may not work as expected if you run the Flask server on Windows. This is just a rough prototype of what we can support.

Reply:
We also want to be able to handle the case where users put their plugin tasks in an `__init__.py` file.

Reply:
Got it. Yeah, I wasn't considering how nested folders were implemented. From your comment, it sounds like this is handled by putting the full path in the filename.